File: app/src/main/java/com/numero/sojodia/repository/IBusDataRepository.java
package com.numero.sojodia.repository;

import com.numero.sojodia.model.BusDataFile;
import com.numero.sojodia.model.BusTime;

import java.util.List;

import io.reactivex.Observable;

public interface IBusDataRepository {

    Observable<List<BusTime>> loadBusData(BusDataFile busDataFile);

}

File: app/src/main/kotlin/com/numero/sojodia/repository/IBusDataRepository.kt
package com.numero.sojodia.repository

import com.numero.sojodia.model.BusTime
import com.numero.sojodia.model.Config
import com.numero.sojodia.resource.datasource.api.response.BusDataResponse
import io.reactivex.Observable

interface IBusDataRepository {

    val tkBusTimeListGoing: List<BusTime>

    val tkBusTimeListReturn: List<BusTime>

    val tndBusTimeListGoing: List<BusTime>

    val tndBusTimeListReturn: List<BusTime>

    val isNoBusTimeData: Boolean

    fun loadBusDataConfig(): Observable<Config>

    fun loadAndSaveBusData(): Observable<BusDataResponse>

    fun reloadBusData()

}
